p2p/net/conn.go

```diff
@@ -108,6 +111,17 @@ func (c *FormattedConnection) RemoteAddr() net.Addr {
 	return c.remoteAddr
 }
 
+// RemoteListenPort this is used to know on which port the peer from this connection listens
+func (c *FormattedConnection) RemoteListenPort() uint16 {
```

Comment:
receiver can be of non * type

Reply:
then shouldn't all other methods be non-pointer types? apart from mutating methods ofcourse